SYCL/SubGroupMask/Basic.cpp

```diff
@@ -0,0 +1,104 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// REQUIRES: gpu
+// UNSUPPORTED: cuda, hip
+// GroupNonUniformBallot capability is supported on Intel GPU only
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+//==---------- Basic.cpp - sub-group mask basic test -----------*- C++ -*---==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+using namespace sycl;
+constexpr int global_size = 128;
+constexpr int local_size = 32;
+int main() {
+#ifdef SYCL_EXT_ONEAPI_SUB_GROUP_MASK
+  queue Queue;
+
+  try {
+    nd_range<1> NdRange(global_size, local_size);
+    int Res = 0;
+    {
+      buffer resbuf(&Res, range<1>(1));
+
+      Queue.submit([&](handler &cgh) {
+        auto resacc = resbuf.get_access<access::mode::read_write>(cgh);
+
+        cgh.parallel_for<class sub_group_mask_test>(
+            NdRange, [=](nd_item<1> NdItem) [[intel::reqd_sub_group_size(32)]] {
+              size_t GID = NdItem.get_global_linear_id();
+              auto SG = NdItem.get_sub_group();
+              // AAAAAAAA
+              auto gmask_gid2 =
+                  ext::oneapi::group_ballot(NdItem.get_sub_group(), GID % 2);
+              // B6DB6DB6
+              auto gmask_gid3 =
+                  ext::oneapi::group_ballot(NdItem.get_sub_group(), GID % 3);
+
+              if (!GID) {
+                int res = 0;
+
+                for (size_t i = 0; i < SG.get_max_local_range()[0]; i++) {
+                  res |= !((gmask_gid2 | gmask_gid3)[i] == (i % 2 || i % 3))
+                         << 1;
+                  res |= !((gmask_gid2 & gmask_gid3)[i] == (i % 2 && i % 3))
+                         << 2;
+                  res |= !((gmask_gid2 ^ gmask_gid3)[i] ==
+                           ((bool)(i % 2) ^ (bool)(i % 3)))
+                         << 3;
+                }
+                gmask_gid2 <<= 8;
+                uint32_t r = 0;
+                gmask_gid2.extract_bits(r);
+                res |= (r != 0xaaaaaa00) << 4;
+                (gmask_gid2 >> 4).extract_bits(r);
+                res |= (r != 0x0aaaaaa0) << 5;
+                gmask_gid3.insert_bits((char)0b01010101, 8);
+                res |= (!gmask_gid3[8] || gmask_gid3[9] || !gmask_gid3[10] ||
+                        gmask_gid3[11])
+                       << 6;
+                marray<unsigned char, 6> mr{1};
```

Comment:
Sorry if this is obvious, but why is this a 6?

Reply:
Just longer-than-sub_group_mask-can accept.